File: src/main/java/com/gmail/samueler53/fwhumanstratego/listeners/onPlayerChattingListener.java
package com.gmail.samueler53.fwhumanstratego.listeners;

import com.gmail.samueler53.fwhumanstratego.managers.GameManager;
import com.gmail.samueler53.fwhumanstratego.objects.Game;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.AsyncPlayerChatEvent;

import java.util.Objects;
import java.util.UUID;

public class onPlayerChattingListener implements Listener {

    final GameManager gameManager = GameManager.getInstance();

    @EventHandler
    public void onPlayerChatting(AsyncPlayerChatEvent event) {
        Player player = event.getPlayer();
        UUID uuid = player.getUniqueId();
        if (gameManager.isPlayerPlaying(uuid).isPresent()) {
            Game game = gameManager.getGameWherePlayerPlaying(uuid);
            if (game.hasASquad(uuid)) {
                event.setCancelled(true);
                if (game.getRed().getPlayersRoles().containsKey(uuid)) {
                    for (UUID uuidPlayersInRedTeam : game.getRed().getPlayersRoles().keySet()) {
                        Objects.requireNonNull(Bukkit.getPlayer(uuidPlayersInRedTeam)).sendMessage(ChatColor.DARK_RED + "Red team " + player.getDisplayName() + ": " + event.getMessage());
                    }
                } else if (game.getBlue().getPlayersRoles().containsKey(uuid)) {
                    for (UUID uuidPlayersInBlueTeam : game.getBlue().getPlayersRoles().keySet()) {
                        Objects.requireNonNull(Bukkit.getPlayer(uuidPlayersInBlueTeam)).sendMessage(ChatColor.BLUE + "Blue team " + player.getDisplayName() + ": " + event.getMessage());
                    }
                }
            }
        }
    }
}

File: src/main/java/com/gmail/samueler53/fwhumanstratego/listeners/OnPlayerChattingListener.kt
package com.gmail.samueler53.fwhumanstratego.listeners

import com.gmail.samueler53.fwhumanstratego.managers.GameManager
import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.AsyncPlayerChatEvent

class OnPlayerChattingListener : Listener {

    @EventHandler
    fun onPlayerChatting(event: AsyncPlayerChatEvent) {
        val game = GameManager.getGameForPlayer(event.player) ?: return
        if (!game.isPlayerInTeam(event.player)) return
        event.isCancelled = true
        val team = game.getTeamForPlayer(event.player)
        val message = when (team) {
            game.redTeam -> "§4Red team %s: %s"
            game.blueTeam -> "§9Blue team %s: %s"
            else -> return
        }
        for (p in team.playersRoles.keys.mapNotNull(Bukkit::getPlayer)) {
            p.sendMessage(message.format(event.player.displayName, event.message))
        }
    }
}